ziti/enroll/enroll.go

```diff
@@ -380,7 +380,18 @@ func enrollCAAuto(enFlags EnrollmentFlags, cfg *config.Config, caPool *x509.Cert
 			if resp.StatusCode == http.StatusConflict {
 				return errors.New("the provided identity has already been enrolled")
 			} else {
-				return errors.Errorf("enroll error: %s", resp.Status)
+				body, err := ioutil.ReadAll(resp.Body)
+				if err != nil {
+					return errors.Errorf("enroll error: %s", resp.Status)
+				}
+
+				if respContainer, err := gabs.ParseJSON(body); err == nil {
+					code := respContainer.Path("error.code").Data().(string)
+					message := respContainer.Path("error.message").Data().(string)
+					return errors.Errorf("enroll error: %s: %s: %s", resp.Status, code, message)
+				} else {
+					return errors.Errorf("enroll error: %s", resp.Status)
```

Comment:
would it make sense to write the body out here too as a debug 'just in case'?

Reply:
Done